tensorflow/compiler/xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -3059,5 +3103,166 @@ Status IrEmitterUnnested::EmitConstantGlobals() {
   return Status::OK();
 }
 
+// Overall, emit code for slices based on the below structure. A `slice_guard_i`
+// and a `slice_i` are generated for each slice. `slice_guard_i` computes
+// the guarding condition to decide if it should jump into `slice_i`
+// for writing to slice output or continue to next `slice_guard_{i+1}`.
+//
+// init_block:
+//   Compute values of slice input operands
+//   Br slice_guard_0
+//
+// slice_guard_0:
+//   Compute guarding_cond0
+//   CondBr guarding_cond0, slice_0, slice_guard_1
+//
+// slice_0:
+//   Write to output of slice0
+//   Br slice_guard_1
+//
+// slice_guard_1:
+//   Compute guarding_cond1
+//   CondBr guarding_cond1, slice_1, exit_block
+//
+// slice_1:
+//   Write to output of slice1
+//   Br exit_block
+//
+// exit_block:
+//
+void IrEmitterUnnested::EmitElementForInputFusibleSlices(
+    HloInstruction* unnested_hlo, const llvm_ir::IrArray::Index& index) {
+  VLOG(10) << "Emitting slice input fusion for " << unnested_hlo->ToString();
+
+  HloInstruction* slice_or_tuple = unnested_hlo->fused_expression_root();
+  auto slice_instructions = ([&]() -> absl::Span<HloInstruction* const> {
```

Comment:
Two pairs of parens are not required: `auto x = [&] { ... }()` also works.

Reply:
Good suggestion.